client/jetpack-cloud/sections/manage/pricing/index.web.js

```diff
@@ -1,11 +1,14 @@
 import page from '@automattic/calypso-router';
 import { makeLayout, render as clientRender } from 'calypso/controller';
-import { jetpackManagePricingPath } from 'calypso/lib/jetpack/paths';
+import { setLocaleMiddleware } from 'calypso/controller/shared';
 import * as controller from './controller';
 
 export default function () {
+	page( '/manage/pricing', controller.jetpackManagePricingContext, makeLayout, clientRender );
+
 	page(
-		jetpackManagePricingPath(),
+		'/:lang/manage/pricing',
```

Comment:
`getLanguageRouteParam()` can be used here as well, and since it defines the parameter as optional by default, defining the route above without the param will no longer be needed.

Reply:
I'm curious how this works (probably some middleware magic), as the route regex itself wouldn't match both routes given the language URLs have an extra slash vs. the default, but it does work: 244a1f86fa

Reply:
When you have an optional parameter in the path, `@automattic/calypso-router` (i.e. page()) will build the regex matcher by [grouping the parameter pattern along with the preceding forward slash and apply optional quantifier to the whole group](https://github.com/Automattic/wp-calypso/blob/trunk/packages/calypso-router/src/index.js#L320-L325).

So for example, `/:lang(en|es|fr)?/manage/pricing` will end up being compiled to something like `/^(?:\/(en|es|fr))?\/manage\/pricing/?$/`.